### PR TITLE
Add Cloud Functions support

### PR DESCRIPTION
**Pull request by kroikie**
Add plugin that allows use of Firebase Functions callable API.

### CODE REVIEW

File: packages/cloud_functions/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.DS_Store
```
Comment (bparrishMines) [inline packages/cloud_functions/.gitignore]: I think you can delete all the `.gitignore` files. I think we are trying to keep only one from now on from https://github.com/flutter/plugins/pull/588

File: packages/cloud_functions/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## 0.0.1
+
+* Initial release
```
Comment (bparrishMines) [inline packages/cloud_functions/CHANGELOG.md]: Could you add more description on what is in the initial release?

File: packages/cloud_functions/android/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+*.iml
```
Comment (bparrishMines) [inline packages/cloud_functions/android/.gitignore]: Same as above .gitignore

File: packages/cloud_functions/example/android/.gitignore [hunk 4]
```diff
@@ -0,0 +1,10 @@
+*.iml
```
Comment (bparrishMines) [inline packages/cloud_functions/example/android/.gitignore]: Same as above .gitignore

File: packages/cloud_functions/example/ios/.gitignore [hunk 5]
```diff
@@ -0,0 +1,45 @@
+.idea/
```
Comment (bparrishMines) [inline packages/cloud_functions/example/ios/.gitignore]: Same as above .gitignore

File: packages/cloud_functions/example/lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+                new MaterialButton(
+                  child: const Text('SEND REQUEST'),
+                  onPressed: () async {
+                    try {
+                      final dynamic resp = await CloudFunctions.instance.call(
+                          functionName: 'repeat',
+                          parameters: <String, dynamic>{
+                            'message': 'hello world!',
+                            'count': _responseCount,
+                          });
```
Comment (bparrishMines) [inline packages/cloud_functions/example/lib/main.dart]: nit: comma after the '}'

File: packages/cloud_functions/ios/.gitignore [hunk 7]
```diff
@@ -0,0 +1,36 @@
+.idea/
```
Comment (bparrishMines) [inline packages/cloud_functions/ios/.gitignore]: Same as above .gitignore

File: packages/cloud_functions/.gitignore (hunk 1 above)
Comment (kroikie) [reply packages/cloud_functions/.gitignore]: Done

File: packages/cloud_functions/CHANGELOG.md (hunk 2 above)
Comment (kroikie) [reply packages/cloud_functions/CHANGELOG.md]: Done

File: packages/cloud_functions/example/lib/main.dart (hunk 6 above)
Comment (kroikie) [reply packages/cloud_functions/example/lib/main.dart]: Done

File: packages/cloud_functions/ios/.gitignore (hunk 7 above)
Comment (kroikie) [reply packages/cloud_functions/ios/.gitignore]: Done

File: packages/cloud_functions/example/ios/.gitignore (hunk 5 above)
Comment (kroikie) [reply packages/cloud_functions/example/ios/.gitignore]: Done

File: packages/cloud_functions/example/android/.gitignore (hunk 4 above)
Comment (kroikie) [reply packages/cloud_functions/example/android/.gitignore]: Done

File: packages/cloud_functions/example/.gitignore [hunk 8]
```diff
@@ -0,0 +1,9 @@
+.DS_Store
```
[the remark this answers is not in the document]
Comment (kroikie) [reply packages/cloud_functions/example/.gitignore]: Done

File: packages/cloud_functions/android/.gitignore (hunk 3 above)
Comment (kroikie) [reply packages/cloud_functions/android/.gitignore]: Done

**Pull request merged**